db_stress_tool/expected_state.cc

```diff
@@ -264,12 +285,17 @@ Status FileExpectedStateManager::Open() {
 
   std::string expected_state_file_path =
       GetPathForFilename(kLatestBasename + kStateFilenameSuffix);
+  std::string expected_persisted_seqno_file_path = GetPathForFilename(
+      kPersistedSeqnoBasename + kPersistedSeqnoFilenameSuffix);
   bool found = false;
   if (s.ok()) {
     Status exists_status = Env::Default()->FileExists(expected_state_file_path);
     if (exists_status.ok()) {
       found = true;
     } else if (exists_status.IsNotFound()) {
+      assert(Env::Default()
```

Comment:
1. Do we need to update `Clean()` to handle the new `PERSIST` file? I don't think so because we don't have a notion of a "stale" `PERSIST` file (unlike the state/trace files)
2. I think this check should be correct because we create the expected state file before we create the persisted sequence number file. I am thinking we could break this invariant if we happened to delete the `PERSIST` file before deleting the `LATEST` file, but I don't see us doing that anywhere.

Reply:
> Do we need to update Clean() to handle the new PERSIST file? I don't think so because we don't have a notion of a "stale" PERSIST file (unlike the state/trace files)

Right - PERSIST file is like LATEST file.  LATEST file is excluded from this `filename.rfind(kLatestBasename, 0) == std::string::npos`

> I think this check should be correct because we create the expected state file before we create the persisted sequence number file. I am thinking we could break this invariant if we happened to delete the PERSIST file before deleting the LATEST file, but I don't see us doing that anywhere.

This is really there to simplify the reasoning with no real reason why `expected_persisted_seqno_file_path` can't exist. The ideal way is to put these two into one file but I haven't figured out how to mmap a complicated object like this. Will note as a TODO

Reply:
> The ideal way is to put these two into one file but I haven't figured out how to mmap a complicated object like this.

Yeah, I was also thinking about whether we could combine these 2 files into 1, but it seemed like it could add other unwanted complexity